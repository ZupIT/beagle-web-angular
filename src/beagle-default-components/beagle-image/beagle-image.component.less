/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

figure {
  width: 100%;
  height: 100%;
  margin: 0px;

  &.fit-xy,
  &.fit-center,
  &.center-crop,
  &.center {
    img {
      width: inherit;
      height: inherit;
    }
  }

  &.fit-xy {
    object-fit: fill
  }

  &.fit-center {
    object-fit: cover;
  }

  &.center-crop {
    object-fit: none;
  }

  &.center {
    object-fit: contain;
  }

  img {
    width: inherit;
  }
}