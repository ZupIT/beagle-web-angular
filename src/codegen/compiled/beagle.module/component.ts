/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Type } from '@angular/core'
import {
  kebabToCamelCase,
  removeExtraIndentation,
  replaceToUnderline,
} from '../../utils/formatting'
import { getComponentAnnotations } from '../../utils/metadata'
import { remoteViewSelector } from '../../../constants'


function createViewChildString(name: string, templateName: string, angularVersion: number): string {
  const componentName = replaceToUnderline(name)

  if (angularVersion >= 8) {
    return `'${componentName}': new ViewChild('${templateName}', { static: true })`
  } else {
    return `'${componentName}': new ViewChild('${templateName}')`
  }
}

function createQueries(components: Record<string, Type<any>>, angularVersion: number) {
  const componentNames = Object.keys(components)

  return componentNames.map((name) => {
    const selector = getComponentAnnotations(components[name]).selector
    const templateName = kebabToCamelCase(selector)
    return createViewChildString(name, templateName, angularVersion)
  })
}


export function createComponentString(
    components: Record<string, Type<any>>,
    angularVersion: number,
  ) {
  const queries = createQueries(components, angularVersion)

  // todo: legacy code. Remove the input "loadParams" with v2.0.
  const componentString = `
    @Component({
      selector: '${remoteViewSelector}',
      template,
      inputs: ['loadParams', 'route', 'networkOptions', 'controllerId'],
      queries: {
        ${queries.join(',\n        ')},
      },
    })
    export class BeagleRemoteView extends AbstractBeagleRemoteView {
      constructor(
        beagleProvider: BeagleProvider,
        ngZone: NgZone,
        changeDetector: ChangeDetectorRef,
      ) {
        // @ts-ignore
        super(beagleProvider, ngZone, changeDetector)
      }
    }
  `

  return removeExtraIndentation(componentString, 4)
}
