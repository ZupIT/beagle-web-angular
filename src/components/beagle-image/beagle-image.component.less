/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

figure {
  width: inherit;
  height: inherit;
  min-width: inherit;
  min-height: inherit;
  max-width: inherit;
  max-height: inherit;
  border-radius: inherit;
  margin: 0px;

  .FIT_XY,
  .FIT_CENTER,
  .CENTER_CROP,
  .CENTER {
    img {
      width: inherit;
      height: inherit;
    }
  }

  .FIT_XY {
    object-fit: fill
  }

  .FIT_CENTER {
    object-fit: cover;
  }

  .CENTER_CROP {
    object-fit: none;
  }

  .CENTER {
    object-fit: contain;
  }

  img {
    width: inherit;
    height: inherit;
    min-width: inherit;
    min-height: inherit;
    max-width: inherit;
    max-height: inherit;
    border-radius: inherit;
  }
}
