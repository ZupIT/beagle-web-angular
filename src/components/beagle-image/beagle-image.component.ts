/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Component, Input, OnInit } from '@angular/core'
import { BeagleImageConfig } from '../../runtime/BeagleImageConfig'
import { ImageContentMode } from '../types'

@Component({
  selector: 'beagle-image',
  templateUrl: './beagle-image.component.html',
  styleUrls: ['./beagle-image.component.less'],
})
export class BeagleImageComponent implements OnInit {

  @Input() name = ''
  @Input() path = ''
  @Input() contentMode?: ImageContentMode = 'FIT_CENTER'
  imageName = ''

  ngOnInit() {
    this.imageName = this.name ? 
      BeagleImageConfig.imagesLocation + this.name : this.path

    if (!this.contentMode) {
      this.contentMode = 'FIT_CENTER'
    }
  }

}
