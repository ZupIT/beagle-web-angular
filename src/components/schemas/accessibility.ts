export interface Accessibility {
  accessible: boolean,
  accessibilityLabel?: string,
  isHeader?: boolean,
}
