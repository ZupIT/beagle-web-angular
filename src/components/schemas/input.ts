/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Properties as CSSProperties } from 'csstype'
import { InputHandler } from './input-handler'

export type InputType = 'DATE' | 'EMAIL' | 'PASSWORD' | 'NUMBER' | 'TEXT'

export interface BeagleInputInterface {
  value?: string,
  placeholder?: string,
  disabled?: boolean,
  readOnly?: boolean,
  type?: InputType,
  hidden?: boolean,
  styleId?: string,
  style?: CSSProperties,
  onChange?: InputHandler,
  onBlur?: InputHandler,
  onFocus?: InputHandler,
  error?: string,
  showError?: boolean,
}
