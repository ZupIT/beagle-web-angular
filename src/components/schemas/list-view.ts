/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { BeagleUIElement } from '@zup-it/beagle-web'
import { Properties as CSSProperties } from 'csstype'

export type Direction = 'VERTICAL' | 'HORIZONTAL'

export interface BeagleFutureListViewInterface {
  direction: Direction,
  dataSource: any[],
  iteratorName?: string,
  onInit?: () => void,
  onScrollEnd?: () => void,
  scrollEndThreshold?: number,
  template: BeagleUIElement,
  useParentScroll?: boolean,
  style?: CSSProperties,
}

export interface BeagleListViewInterface {
  direction: Direction,
}
