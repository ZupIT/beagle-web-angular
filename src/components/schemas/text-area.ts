/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Properties as CSSProperties } from 'csstype'

export type InputHandler = (event: { value: string }) => void

export interface TextAreaInterface {
  value?: string,
  label?: string,
  name?: string,
  onChange?: InputHandler,
  onFocus?: InputHandler,
  onBlur?: InputHandler,
  disabled?: boolean,
  readonly?: boolean,
  styleId?: string,
  style?: CSSProperties,
}
