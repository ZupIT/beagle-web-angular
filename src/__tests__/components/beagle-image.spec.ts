/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async } from '@angular/core/testing'
import { BeagleImageComponent } from '../../components/beagle-image/beagle-image.component'
import { Accessibility } from '../../components/schemas/accessibility'
import { ImagePath, ImagePathMode, ImageMode } from '../../components/schemas/image'
import { buildBeagleTestModuleMetadata } from './test-module-builder'

let component: BeagleImageComponent
const imagePathModeMock: ImagePathMode = 'local'
const pathMock: ImagePath = {
  _beagleImagePath_: imagePathModeMock,
  url: 'http://teste.com.br',

}
const accessibilityMock: Accessibility = {
  accessible: true,
  accessibilityLabel: 'Alt text',
}
const imageModeMock: ImageMode = 'FIT_CENTER'

describe('BeagleImageComponent', () => {

  beforeEach(async(() => {
    TestBed.configureTestingModule(buildBeagleTestModuleMetadata([BeagleImageComponent]))
      .compileComponents()

    const fixture = TestBed.createComponent(BeagleImageComponent)
    component = fixture.componentInstance
    component.path = pathMock
    component.accessibility = accessibilityMock
    spyOn(component, 'ngOnChanges').and.callThrough()
  }))

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should add imageSource', () => {
    expect(component.imageSource).toEqual('')
    component.ngOnChanges()
    expect(component.imageSource).toEqual('http://teste.com.br')
  })

  it('should add default mode', () => {
    delete component.mode
    component.ngOnChanges()
    expect(component.mode).toEqual(imageModeMock)
  })
})
