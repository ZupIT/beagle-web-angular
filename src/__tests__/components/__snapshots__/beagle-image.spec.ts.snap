// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleImageComponent should match snapshot 1`] = `
BeagleImageComponent {
  "accessibility": Object {
    "accessibilityLabel": "Alt text",
    "accessible": true,
  },
  "imageSource": "",
  "mode": "FIT_CENTER",
  "ngOnChanges": [Function],
  "path": Object {
    "_beagleImagePath_": "local",
    "url": "http://teste.com.br",
  },
}
`;
